client/packages/system/src/Encounter/DetailView/Toolbar.tsx

```diff
@@ -230,9 +230,19 @@ export const Toolbar: FC<ToolbarProps> = ({
                     onChange={date => {
                       const startDatetime = DateUtils.formatRFC3339(date);
                       setStartDatetime(startDatetime);
+                      const endDt = DateUtils.getDateOrNull(endDatetime);
                       onChange({
                         startDatetime,
-                        endDatetime: endDatetime ?? undefined,
+                        endDatetime: endDatetime
+                          ? DateUtils.formatRFC3339(
```

Comment:
Nit: refactor this into a `endTimeFromStartDate(startDate: string | undefined, endTime: Date)`?

Also check for endDt `endDt ? DateUtils...` then you don't need the null check later...